docs/adoc/microfrontend-platform-developer-guide/chapters/core-concepts/cross-application-communication/intent-based-messaging.adoc

```diff
@@ -23,9 +22,9 @@ In this Chapter
 [[chapter:intent-based-messaging:what-is-intent-based-messaging]]
 [discrete]
 === What is Intent-Based Messaging?
-Intent-based messaging enables controlled collaboration between micro applications, a mechanism known from Android development where an application can start an _Activity_ via an _Intent_ (such as sending an email).
+Intent-based messaging enables controlled collaboration between micro applications, a mechanism known from Android development where an application can start an activity via an _Intent_ (such as sending an email).
 
-This kind of communication is similar to the <<chapter:topic-based-messaging,topic-based>> communication, thus implements also the publish-subscribe messaging pattern, but additionally requires the sending application to declare an intention in its manifest. Unlike topic-based communication, the message (also called the intent) is exclusively transported to micro applications that provide a fulfilling capability through their manifest.
+This kind of communication is similar to the <<chapter:topic-based-messaging,topic-based>> communication, thus implements also the publish-subscribe messaging pattern, but additionally requires the sending application to express an intention in its manifest. Unlike topic-based communication, the message (also called the intent) is exclusively transported to micro applications that provide a fulfilling capability through their manifest.
```

Comment:
I think it is not quite clear from developer perspective when to use topic-based communication and when intent-based. Something like best practices. e.g.
- intent-based communication MUST be used between microfrontends, because of explicit dependencies
- topic-based communication COULD be used  only for microfrontend-internal communication.

Btw. what is the primary use case for topic-based communication? Maybe the platform should even enforce communication between microfrontends via intents - e.g. prevent bad design practices?

Reply:
Good point @k-genov. We should be more clear about when to use intents vs messages.
I would not enforce a specific API because there are use cases for both models. In our project, for cross-application communication, we stick to intent-based communication in order to explicitly manifest the dependencies between the applications. However, that is just our best practice.